### DISCUSSION TITLE
kernel panic when using memory limit

### DISCUSSION THREAD

**Discussion by nekolyshe** [containers/podman#19791] [Q&A]
Hello, I have an issue when running a container with a memory limit.

I used the next commands for clean reproduction (set limit as 5M and write the files in infinite loop):

works as expected:
podman run -dt --name=container1 --memory=5m  --network=none docker.io/arm64v8/busybox:1.35.0 sh -c 'i=0; while true; do echo $i; MEMBLOB$i=$(dd if=/dev/urandom bs=1k count=1024); i=$((i+1)); sleep 1; done' > /dev/null

system fails with attached logs:
podman run -dt --name=container1 --memory=5m --network=none docker.io/arm64v8/busybox:1.35.0 sh -c 'i=0; while true; do echo $i; MEMBLOB$i=$(dd if=/dev/urandom bs=5k count=1024); i=$((i+1)); sleep 1; done' > /dev/null
![screenshot-1](https://github.com/containers/podman/assets/38406695/547772be-4021-4a1a-81a0-80e5f692effa)

the difference between the commands is that in the first case dd command writes files with size 1M and in the second case writes files with size 5M(equals or higher than the mem limit).

Could someone advise what can be the issue?
[podmanInfo.txt](https://github.com/containers/podman/files/12463562/podmanInfo.txt)

Thanks.

**Comment by rhatdan** [comment]
Kernel panics should be reported to whoever provides your kernel or to the upstream.  Podman does not effect this.

Looks like something in the cgroups subsystem is blowing up.

**Reply by Serebana** [reply]
Thank you for the reply, but the issue is that we have enough RAM, and memory limitation works as expected when we fill it continuously but does reboot when we fill the full limit of the container memory by one shot. 
Yes, you're right, looks like we have a wrong correlation between podman and cgroups. 
Do you know how can we trace what goes wrong?
